### ISSUE TITLE
Change `lr_scheduler` config structure

### ISSUE THREAD

**Issue by gitttt-1234** [talmolab/sleap-nn#153]
Currently, the learning rate scheduler config structure is as follows:
```
trainer_config:
    lr_scheduler:
        scheduler: # str -> one of ["StepLR", "ReduceLROnPlateau"]
        step_lr:
        reduce_lr_on_plateau:
```
The `scheduler` parameter seems redundant. Ideally, the scheduler which has a non-empty config should be chosen (similar to how `backbone_config` and `head_configs` are deisgned). The new structure should remove the `scheduler` key from `lr_scheduler` config.
